src/BinSkim.Driver/RoslynAnalyzer/ILOperations.cs

```diff
@@ -0,0 +1,485 @@
+        {
+            Label = label;
+        }
+
+        public ILabelSymbol Label { get; }
+        public override OperationKind Kind => OperationKind.LabelStatement;
+    }
+
+    internal sealed class IfStatement : Statement, IIfStatement
+    {
+        public IfStatement(IExpression condition, IStatement ifTrue)
+        {
+            Condition = condition;
+            IfTrue = ifTrue;
+        }
+        public IExpression Condition { get; }
+        public IStatement IfTrue { get; }
+        public override OperationKind Kind => OperationKind.IfStatement;
+
+        IStatement IIfStatement.IfFalse => null; // we always goto on true and fallthrough on false. 
+    }
+
+    internal sealed class GoToStatement : Statement, IBranchStatement
+    {
+        public GoToStatement(ILabelSymbol target)
+        {
+            Target = target;
+        }
+
+        public ILabelSymbol Target { get; }
+
+        // FEEDBACK: first non 1:1 I see between interface name and kind, others have distinct subkinds. Should this be BranchKind.GoTo and OperationKind.BranchStatement?
+        public override OperationKind Kind => OperationKind.GoToStatement; 
+
+    }
+
+    internal sealed class ThrowStatement : Statement, IThrowStatement
+    {
+        public ThrowStatement(IExpression thrown)
+        {
+            Thrown = thrown;
+        }
+
+        public IExpression Thrown { get; }
+        public override OperationKind Kind => OperationKind.ThrowStatement;
+    }
+
+    internal sealed class ReturnStatement : Statement, IReturnStatement
+    {
+        public ReturnStatement(IExpression returned)
+        {
+            Returned = returned;
+        }
+
+        public IExpression Returned { get; }
+        public override OperationKind Kind => OperationKind.ReturnStatement;
+    }
+
+    internal sealed class TryStatement : Statement, ITryStatement
+    {
+        public TryStatement(IBlockStatement body, ImmutableArray<ICatch> catches, IBlockStatement finallyHandler)
+        {
+            Body = body;
+            Catches = catches;
+            FinallyHandler = finallyHandler;
+        }
+
+        public IBlockStatement Body { get; }
+        public ImmutableArray<ICatch> Catches { get; }
+        public IBlockStatement FinallyHandler { get; }
+        public override OperationKind Kind => OperationKind.TryStatement;
+    }
+
+    internal sealed class Catch : Operation, ICatch
+    {
+        public Catch(ITypeSymbol caughtType, ILocalSymbol exceptionLocal, IExpression filter, IBlockStatement handler)
+        {
+            CaughtType = caughtType;
+            ExceptionLocal = exceptionLocal;
+            Filter = filter;
+            Handler = handler;
+        }
+
+        public ITypeSymbol CaughtType { get; }
+        public ILocalSymbol ExceptionLocal { get; }
+        public IExpression Filter { get; }
+        public IBlockStatement Handler { get; }
+
+        // FEEDBACK: inconsistently named vs. interface
```

Comment:
Yes, these should be reconciled. Changing the name of the interface to "ICatchHandler" is probably best.